### ISSUE TITLE
Inconsistent results when parsing different time conventions (12-hour vs 24-hour clock)

### ISSUE THREAD

**Issue by pawelkrol** [mojombo/chronic#325]
There is a strange behaviour observed when parsing plain times using `chronic` gem:

* When time is expressed in the 12-hour clock convention, we get today's date as a result.
* When time is expressed in the 24-hour clock convention, we get tomorrow's date as a result.

It seems a bit inconsistent to me, please see the following example:

```
2.1.5 :001 > require 'chronic'
 => true 
2.1.5 :002 > Chronic.parse("08:00AM")
 => 2016-02-24 08:00:00 +0100 
2.1.5 :003 > Chronic.parse("08:00")
 => 2016-02-25 08:00:00 +0100 
```

It has been confirmed with `0.9.1` and `0.10.2` releases, as well as the latest `master` branch.

**Comment by davispuh** [comment]
yeah, it's a bug, it's fixed in my `rewrite` branch but that's not ready yet and will be released as next major version.

**Comment by pawelkrol** [comment]
Thanks a lot for your confirmation. I am looking forward to a new release version containing this fix.
